api/repositories/service_instance_repository.go

```diff
@@ -274,7 +294,40 @@ func (r *ServiceInstanceRepo) servicePlanVisible(ctx context.Context, planGUID s
 	return slices.Contains(servicePlan.Spec.Visibility.Organizations, space.Namespace), nil
 }
 
-func (r *ServiceInstanceRepo) PatchServiceInstance(ctx context.Context, authInfo authorization.Info, message PatchServiceInstanceMessage) (ServiceInstanceRecord, error) {
+func (r *ServiceInstanceRepo) PatchUserProvidedServiceInstance(ctx context.Context, authInfo authorization.Info, message PatchUPSIMessage) (ServiceInstanceRecord, error) {
+	cfServiceInstance := &korifiv1alpha1.CFServiceInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: message.SpaceGUID,
+			Name:      message.GUID,
+		},
+	}
+	if err := r.klient.Get(ctx, cfServiceInstance); err != nil {
+		return ServiceInstanceRecord{}, apierrors.FromK8sError(err, ServiceInstanceResourceType)
+	}
+
+	err := r.klient.Patch(ctx, cfServiceInstance, func() error {
+		message.Apply(cfServiceInstance)
+		return nil
+	})
+	if err != nil {
+		return ServiceInstanceRecord{}, apierrors.FromK8sError(err, ServiceInstanceResourceType)
+	}
+
+	if message.Credentials != nil {
+		cfServiceInstance, err = r.migrateLegacyCredentials(ctx, cfServiceInstance)
+		if err != nil {
+			return ServiceInstanceRecord{}, err
+		}
+		err = r.patchCredentialsSecret(ctx, cfServiceInstance, *message.Credentials)
+		if err != nil {
+			return ServiceInstanceRecord{}, apierrors.FromK8sError(err, ServiceInstanceResourceType)
+		}
+	}
+
+	return cfServiceInstanceToRecord(*cfServiceInstance), nil
+}
+
+func (r *ServiceInstanceRepo) PatchManagedServiceInstance(ctx context.Context, authInfo authorization.Info, message PatchManagedSIMessage) (ServiceInstanceRecord, error) {
```

Comment:
If I recall correctly, credentials make only sense for UPSIs, therefore dealing with the credentials secret makes no sense for managed services. Speaking of that, the patch message for managed services probably does not need the `Credentials` field